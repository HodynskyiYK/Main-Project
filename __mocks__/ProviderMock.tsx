import React from 'react'
import {MemoryRouter} from 'react-router-dom'
import { MovieModalContextProvider } from '../src/context/movie-modal-context'
import { Provider } from 'react-redux'
import { store } from '../src/store'
import { MovieDetailsContextProvider } from '../src/context/movie-details-context'

// @ts-ignore
export const ProvidersMock = ({ children }) => {
    return (
        <Provider store={store}>
            <MovieModalContextProvider>
                <MovieDetailsContextProvider>
                    <MemoryRouter>{children}</MemoryRouter>
                </MovieDetailsContextProvider>
            </MovieModalContextProvider>
        </Provider>
    )
}