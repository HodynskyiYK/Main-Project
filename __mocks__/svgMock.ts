// Mock Svg files
// https://github.com/gregberge/svgr/issues/83#issuecomment-437342861
const SvgMock = "icon-mock"
export const ReactComponent = SvgMock
export default SvgMock