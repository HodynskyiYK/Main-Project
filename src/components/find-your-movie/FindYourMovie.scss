.find-your-movie {
  padding: 2.125rem 3.75rem 8.375rem;

  .h1 {
    margin-bottom: 2.375rem;
  }

  .search-form {}

  .form-group {
    display: flex;
    align-items: center;
    justify-content: flex-start;
  }

  .find-input {
    flex: 1 0 17.625rem;
    max-width: 100%;
    margin-right: 0.875rem;
  }

  .confirm-btn {
    color: $white_color;
    background: $red_color;
    min-width: 14.375rem;
    text-align: center;
    padding: 1.125rem 1.125rem;
  }
}