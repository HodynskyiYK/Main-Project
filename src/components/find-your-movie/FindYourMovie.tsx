import React, {FC, FormEvent, useState} from 'react'

export const FindYourMovie: FC = () => {
    const [searchButtonClick, setSearchButtonClick] = useState<number>(0)

    const formHandler = (event: FormEvent) => {
        event.preventDefault()
        setSearchButtonClick((prev) => prev + 1)
    }

    if (searchButtonClick > 0) {
        throw new Error('error')
    }

    return (
        <div className="find-your-movie">
            <p className="h1">
                Find your movie
            </p>
            <form
                className="search-form"
                onSubmit={formHandler}
            >
                <div className="form-group">
                    <input
                        className="input find-input"
                        type="text"
                        name="find-movie"
                        placeholder="What do you want to watch?"
                    />
                    <button className="btn confirm-btn">Search</button>
                </div>
            </form>
        </div>
    )
}
