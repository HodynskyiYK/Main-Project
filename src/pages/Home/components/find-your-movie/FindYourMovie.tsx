import React, {FC, FormEvent, useState} from 'react'
import classnames from 'classnames'
import { Button, ButtonTypes } from '../../../../components/button'
import { Input } from '../../../../components/input'
import styles from './FindYourMovie.module.scss'

export const FindYourMovie: FC = () => {
    const [searchButtonClick, setSearchButtonClick] = useState<number>(0)

    const formHandler = (event: FormEvent) => {
        event.preventDefault()
        setSearchButtonClick((prev) => prev + 1)
    }

    if (searchButtonClick > 0) {
        throw new Error('error')
    }

    return (
        <div className={styles.findYourMovie}>
            <p className={classnames('h1', styles.h1)}>
                Find your movie
            </p>
            <form
                className={styles.searchForm}
                onSubmit={formHandler}
            >
                <div className={styles.formGroup}>
                    <div className={styles.findInput}>
                        <Input
                            inputName={'find-movie'}
                            inputType={'text'}
                            inputPlaceholder={'What do you want to watch?'}
                        />
                    </div>
                    <Button
                        buttonClassName={ButtonTypes.SEARCH_BTN}
                        buttonText={'Search'}
                    />
                </div>
            </form>
        </div>
    )
}
