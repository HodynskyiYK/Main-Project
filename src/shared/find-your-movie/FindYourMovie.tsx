import React, { FC, FormEvent, useEffect, useState } from 'react'
import {Redirect, useLocation, useHistory} from 'react-router-dom'
import classnames from 'classnames'
import { Button, ButtonTypes } from '../../components/button'
import { Input } from '../../components/input'
import { useQuery } from '../../hooks'
import styles from './FindYourMovie.module.scss'

export const FindYourMovie: FC = () => {
    const [redirectTo, setRedirectTo] = useState<boolean>(false)
    const [value, setValue] = useState<string>('')
    const searchQuery = useQuery()
    const history = useHistory()
    const {pathname} = useLocation()

    useEffect(() => {
        const movieTitle = searchQuery.get('search')

        if (movieTitle && pathname === '/search') {
            setValue(movieTitle)
        }
    // eslint-disable-next-line
    }, [])

    const formHandler = (event: FormEvent) => {
        event.preventDefault()
        if (pathname !== '/search') {
            setRedirectTo(true)
        } else {
            history.push(`/search?search=${value}`)
        }
    }

    if (redirectTo) {
        return (
            <Redirect to={`/search?search=${value}`} />
        )
    }

    return (
        <div className={styles.findYourMovie}>
            <p className={classnames('h1', styles.h1)}>Find your movie</p>
            <form
                className={styles.searchForm}
                onSubmit={formHandler}
            >
                <div className={styles.formGroup}>
                    <div className={styles.findInput}>
                        <Input
                            inputName={'find-movie'}
                            inputType={'text'}
                            inputPlaceholder={'What do you want to watch?'}
                            inputValue={value}
                            onChange={(event) => setValue(event.target.value)}
                        />
                    </div>
                    <Button
                        buttonClassName={ButtonTypes.SEARCH_BTN}
                        buttonText={'Search'}
                    />
                </div>
            </form>
        </div>
    )
}
