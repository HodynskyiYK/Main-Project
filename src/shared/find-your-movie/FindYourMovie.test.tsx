import React from 'react'
import { render } from '@testing-library/react'
import { FindYourMovie } from './FindYourMovie'
import { ProvidersMock } from '../../../__mocks__/ProviderMock'

describe('FindYourMovie tests', () => {

    test('To match snapshot', () => {
        const {asFragment} = render(<ProvidersMock><FindYourMovie/></ProvidersMock>)
        expect(asFragment()).toMatchSnapshot()
    })
})