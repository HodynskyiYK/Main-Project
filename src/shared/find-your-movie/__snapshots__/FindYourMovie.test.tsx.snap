// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FindYourMovie tests To match snapshot 1`] = `
<DocumentFragment>
  <div
    class="findYourMovie"
  >
    <p
      class="h1 h1"
    >
      Find your movie
    </p>
    <form
      class="searchForm"
    >
      <div
        class="formGroup"
      >
        <div
          class="findInput"
        >
          <div
            class="inputField"
          >
            <input
              class="field"
              id="find-movie"
              name="find-movie"
              placeholder="What do you want to watch?"
              type="text"
              value=""
            />
          </div>
        </div>
        <button
          class="btn searchBtn"
        >
          Search
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
